### PR TITLE
feat: [maps-mapsplatformdatasets] new module for maps-mapsplatformdatasets

### PR DESCRIPTION
**Pull request by emmileaf**
Supersedes previous attempt in #9119.
 
Initial generation for new mapsplatformdatasets client library: 

```
python3.9 generation/new_client/new-client.py generate \
  --api_shortname=maps-mapsplatformdatasets \
  --proto-path=google/maps/mapsplatformdatasets \
  --name-pretty="Maps Platform Datasets API" \
  --product-docs="https://developers.google.com/maps/documentation" \
  --api-description="The Maps Platform Datasets API enables developers to ingest geospatially-tied datasets
    that they can use to enrich their experience of Maps Platform solutions (e.g. styling, routing)." \
  --cloud-api=false \
  --distribution-name="com.google.maps:google-maps-mapsplatformdatasets"
```

### CODE REVIEW

File: java-maps-mapsplatformdatasets/.repo-metadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "api_shortname": "maps-mapsplatformdatasets",
+  "name_pretty": "Maps Platform Datasets API",
+  "product_documentation": "https://developers.google.com/maps/documentation",
```
Comment (emmileaf) [inline java-maps-mapsplatformdatasets/.repo-metadata.json:4]: This parent product homepage is a placeholder for the product documentation link which is not yet available, but our script requires something starting with "https://".

File: java-maps-mapsplatformdatasets/google-maps-mapsplatformdatasets/pom.xml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.maps</groupId>
```
Comment (suztomo) [inline java-maps-mapsplatformdatasets/google-maps-mapsplatformdatasets/pom.xml:4]: Group ID looks good.

File: java-maps-mapsplatformdatasets/grpc-google-maps-mapsplatformdatasets-v1alpha/pom.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.maps.api.grpc</groupId>
```
Comment (suztomo) [inline java-maps-mapsplatformdatasets/grpc-google-maps-mapsplatformdatasets-v1alpha/pom.xml:5]: Group ID looks good.

File: java-maps-mapsplatformdatasets/google-maps-mapsplatformdatasets-bom/pom.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.maps</groupId>
```
Comment (suztomo) [inline java-maps-mapsplatformdatasets/google-maps-mapsplatformdatasets-bom/pom.xml:4]: Group ID looks good.

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
"snippet-bot check" fails because this is not Cloud product. It's fine.

**Pull request merged**